packages/EasyCore/src/Bridge/Symfony/Validator/Constraints/AbnValidator.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyCore\Bridge\Symfony\Validator\Constraints;
+
+use EoneoPay\Utils\Exceptions\UnexpectedTypeException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * Validates an Australian Business Number.
+ *
+ * @link https://abr.business.gov.au/Help/AbnFormat
+ */
+final class AbnValidator extends ConstraintValidator
+{
+    /**
+     * @var int
+     */
+    private const ABN_LENGTH = 11;
+
+    /**
+     * @var int
+     */
+    private const MODULUS = 89;
+
+    /**
+     * @var int[]
+     */
+    private const WEIGHTS = [10, 1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
+
+    /**
+     * Validates an Australian Business Number.
+     *
+     * @throws \EoneoPay\Utils\Exceptions\UnexpectedTypeException
+     */
+    public function validate($value, Constraint $constraint): void
+    {
+        if ($constraint instanceof Abn === false) {
+            throw new UnexpectedTypeException($constraint, Abn::class);
+        }
+
+        if ($value === null || $value === '') {
+            return;
+        }
+
+        if (\is_string($value) === false && (\is_object($value) && method_exists($value, '__toString')) === false) {
+            throw new UnexpectedValueException($value, 'string');
+        }
+
+        $abn = (string) $value;
+
+        if (\strlen($abn) !== self::ABN_LENGTH) {
+            $this->context->buildViolation($constraint->message)
+                ->setCode(Abn::INVALID_LENGTH_ERROR)
+                ->addViolation();
+
+            return;
+        }
+
+        if (\ctype_digit($abn) === false || (int)$abn[0] === 0) {
```

Comment:
Why did you add a second check? I think this check does not match an error code we set here: 0 is not a invalid character.

Reply:
First of all, first character cannot be 0 according to documentation. Second, when we make `-1` thing below, the first character becomes `-`, not the `-1` as you could expect. That causes an error when we try to multiply digits by weight, because minus sign `-` is not a digit.

Reply:
Ok, let's introduce `Abn::LEADING_ZERO_ERROR` then to be precise and clear. And move this check to a dedicated if statement.

Reply:
Done.